thriftrw/protocol/binary.pyx

```diff
@@ -334,7 +349,7 @@ class BinaryProtocol(Protocol):
         return buff.value
 
     def deserialize_value(self, typ, s):
-        buff = BytesIO(s)
+        buff = ReadBuffer(s)
```

Comment:
Discussed offline but for the record: The only simple way to go from BytesIO's `bytes` output to primitives like int/bool/doubles is to use `struct.pack` which is sort of what this is sort of trying to avoid int he first place.